tests/e2e/kubetest2-kops/builder/build.go

```diff
@@ -83,21 +85,37 @@ func (b *BuildOptions) Build() (*BuildResults, error) {
 		return results, nil
 	}
 
-	cmd := exec.Command("make", "gcs-publish-ci")
+	var cmd exec.Cmd
+	switch {
+	case strings.HasPrefix(b.StageLocation, "gs://"):
+		cmd = exec.Command("make", "gcs-publish-ci")
+		cmd.SetEnv(
+			fmt.Sprintf("GCS_LOCATION=%v", gcsLocation),
+		)
+	case strings.HasPrefix(b.StageLocation, "s3://"):
+		if b.S3BucketRegion == "" {
+			return nil, errors.New("missing required S3 bucket region")
+		}
+		cmd = exec.Command("make", "s3-publish-ci")
+		cmd.SetEnv(
+			fmt.Sprintf("S3_BUCKET=%v", gcsLocation),
+			fmt.Sprintf("S3_REGION=%v", b.S3BucketRegion),
+		)
+	}
 	cmd.SetEnv(
 		fmt.Sprintf("HOME=%v", os.Getenv("HOME")),
 		fmt.Sprintf("PATH=%v", os.Getenv("PATH")),
-		fmt.Sprintf("GCS_LOCATION=%v", gcsLocation),
 		fmt.Sprintf("GOPATH=%v", os.Getenv("GOPATH")),
 	)
+
 	cmd.SetDir(b.KopsRoot)
 	exec.InheritOutput(cmd)
 	if err := cmd.Run(); err != nil {
 		return nil, err
 	}
 
 	// Get the full path (including subdirectory) that we uploaded to
-	// It is written by gcs-publish-ci to .build/upload/latest-ci.txt
+	// It is written by the *-publish-ci make tasks to .build/upload/latest-ci.txt
```

Comment:
It might be nice to spell out the two tasks here (assuming we can't combine them), to make it more explicit and to help with searching...